### PR TITLE
Made it possible to write byte[] larger then maxWriteBufferSize.

### PR DESCRIPTION
**Pull request by 4dn-oss**
Add some process not to raise 'RingBuffer of length xxxx cannot accomodate XXXX bytes.' exception.
Fix RingBuffer#write() logic when wraparound occurs with non-zero offset.

### CODE REVIEW

**Comment by pepijnve** [comment]
Might be good to add a test case in RingBufferTest for the non-zero offset write.

**Comment by hierynomus** [comment]
Can you submit one or more testcases for this scenario?

**Comment by 4dn-oss** [comment]
OK, I'll try it.

File: src/test/groovy/com/hierynomus/smbj/share/RingBufferTest.groovy [hunk 1]
```diff
@@ -232,6 +232,24 @@ public class RingBufferTest extends Specification {
     byteArray == [5, 6, 7, 8, 9, 10] as byte[]
   }
 
+  def "should be able to append bytes in the middle of array if write position has wrapped around"() {
+	given:
```
Comment (hierynomus) [inline src/test/groovy/com/hierynomus/smbj/share/RingBufferTest.groovy]: Can you apply the right formatting? Looks good otherwise.
